lab_5_scrapper/scrapper.py

```diff
@@ -127,11 +268,43 @@ def _extract_url(self, article_bs: BeautifulSoup) -> str:
         Returns:
             str: Url from HTML
         """
+        links = article_bs.find_all(class_='news-teaser__link')
+        url = ''
+
+        for link in links:
+            url = self.url_pattern + link['href']
+            if url in self.get_search_urls():
+                continue
+            if url and url not in self.urls:
+                break
+
+        return url
+
 
     def find_articles(self) -> None:
         """
         Find articles.
         """
+        seed_urls = self.get_search_urls()
+
+        for seed_url in seed_urls:
+            response = make_request(seed_url, self.config)
+            if not response.ok:
+                continue
+
+            article_soup = BeautifulSoup(response.text, features='html.parser')
+            new_url = self._extract_url(article_soup)
+
+
+            while new_url:
+                if len(self.urls) == self.config.get_num_articles():
+                    break
```

Comment:
infinite loop if not enough articles on the page?

Reply:
Changed it